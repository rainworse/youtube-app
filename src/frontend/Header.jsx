import { Link, Outlet, useNavigate } from 'react-router-dom';
import './styles/Header.css';
import LoginLogoutField from './components/LoginLogoutField';
import { useRef } from 'react';

const Header = () => {
  const searchRef = useRef(null);
  const navigate = useNavigate();

  const search = (e) => {
    e.preventDefault();
    navigate('/search/' + searchRef.current.value);
  };

  return (
    <>
      <div className="header">
        <Link to="/" className="header-logo-anchor">
          <img
            src={window.location.origin + '/public/assets/logo.svg'}
            alt="logo"
            className="header-logo"
          />
        </Link>

        <form className="header-form" onSubmit={search}>
          <input type="text" className="header-search-input" ref={searchRef} />
          <input
            type="submit"
            className="header-search-button"
            value={search}
          />
        </form>

        <LoginLogoutField />
      </div>
      <Outlet />
    </>
  );
};

export default Header;
