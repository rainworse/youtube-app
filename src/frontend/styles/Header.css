.header {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
  margin-bottom: 20px;
  margin-left: 30px;
  margin-right: 30px;
}

.header-logo {
  width: 50px;
}

.header-logo-anchor {
  margin-right: auto;
}

.header-form {
  width: 50%;
  display: flex;
  margin-right: auto;
  justify-content: center;
}

.header-search-input {
  width: 50%;
  border: none;
  border-bottom: 2px solid rgb(243, 243, 243);
  border-right: 2px solid rgb(243, 243, 243);
  border-radius: 10px;
  font-family: 'Tinos', serif;
  font-size: 16px;
  height: 35px;
  padding: 10px;
  box-sizing: border-box;
  margin-right: -35px;
  padding-right: 35px;
}

.header-search-button {
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  width: 25px;
  height: 25px;
  background: url(/public/assets/search.svg) no-repeat;
  background-size: 25px;
  margin-top: auto;
  margin-bottom: auto;
}

.header-login-anchor {
  text-decoration: none;
  color: black;
  font-weight: bold;
}

.header-logout {
  display: flex;
  align-items: center;
}

.header-profile-image {
  width: 35px;
}

.header-profile-text {
  margin-left: 10px;
  margin-right: 10px;
  display: flex;
  flex-direction: column;
}

.header-logout-anchor {
  text-decoration: none;
  color: black;
  font-weight: bold;
  align-self: end;
}
