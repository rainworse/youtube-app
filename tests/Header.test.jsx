import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Header from '../src/frontend/Header';
import UserContext from '../src/frontend/UserContext';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

describe('Header', () => {
  it('renders header with null user', () => {
    const header = render(
      <UserContext.Provider value={{ currentUser: null }}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </UserContext.Provider>
    );
    expect(header.container).toMatchSnapshot();
  });

  it('renders header with current user', () => {
    const header = render(
      <UserContext.Provider value={{ currentUser: 'user' }}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </UserContext.Provider>
    );
    expect(header.container).toMatchSnapshot();
  });

  // it('Login button takes you to login page', async () => {
  //   const user = userEvent.setup();

  //   render(
  //     <UserContext.Provider value={{ currentUser: null }}>
  //       <MemoryRouter>
  //         <Header />
  //       </MemoryRouter>
  //     </UserContext.Provider>
  //   );

  //   const login = screen.getByText('Login');

  //   await user.click(login);

  //   expect(screen.getByPlaceholderText('username')).toBeInTheDocument();
  // });
});
