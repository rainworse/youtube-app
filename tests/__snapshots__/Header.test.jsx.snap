// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Login button takes you to login page 1`] = `
<div>
  <div
    class="header"
  >
    <a
      class="header-logo-anchor"
      href="/"
    >
      <img
        alt="logo"
        class="header-logo"
        src="http://localhost:3000/public/assets/logo.svg"
      />
    </a>
    <form
      class="header-form"
    >
      <input
        class="header-search-input"
        type="text"
      />
      <input
        class="header-search-button"
        type="submit"
        value=""
      />
    </form>
    <div
      class="header-login-logout"
    >
      <div
        class="header-login-link"
      >
        <a
          class="header-login-anchor"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header > renders header with current user 1`] = `
<div>
  <div
    class="header"
  >
    <a
      class="header-logo-anchor"
      href="/"
    >
      <img
        alt="logo"
        class="header-logo"
        src="http://localhost:3000/public/assets/logo.svg"
      />
    </a>
    <form
      class="header-form"
    >
      <input
        class="header-search-input"
        type="text"
      />
      <input
        class="header-search-button"
        type="submit"
        value=""
      />
    </form>
    <div
      class="header-login-logout"
    >
      <div
        class="header-logout"
      >
        <div
          class="header-profile-text"
        >
          <div
            class="header-user-name"
          >
            user
          </div>
          <a
            class="header-logout-anchor"
            href="/logout"
          >
            Logout
          </a>
        </div>
        <img
          alt="profile"
          class="header-profile-image"
          src="http://localhost:3000/public/assets/profile.svg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Header > renders header with null user 1`] = `
<div>
  <div
    class="header"
  >
    <a
      class="header-logo-anchor"
      href="/"
    >
      <img
        alt="logo"
        class="header-logo"
        src="http://localhost:3000/public/assets/logo.svg"
      />
    </a>
    <form
      class="header-form"
    >
      <input
        class="header-search-input"
        type="text"
      />
      <input
        class="header-search-button"
        type="submit"
        value=""
      />
    </form>
    <div
      class="header-login-logout"
    >
      <div
        class="header-login-link"
      >
        <a
          class="header-login-anchor"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
</div>
`;
